pkg/archive/archive.go

```diff
@@ -337,8 +328,11 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 		return fmt.Errorf("Unhandled tar header type %d\n", hdr.Typeflag)
 	}
 
-	if err := os.Lchown(path, hdr.Uid, hdr.Gid); err != nil && Lchown {
-		return err
+	// Lchown is not supported on Windows
+	if runtime.GOOS != "windows" {
```

Comment:
can we avoid inline `runtime.GOOS` checks? In #9113 we moved many of these os.[...] calls to system.[...]? These are easier to miss and handling different platforms will make the code just longer wherever these methods are used.